CONTRIBUTING.md

```diff
@@ -126,6 +126,12 @@ CUDA/GPU requirements:
 
 You can obtain CUDA from [https://developer.nvidia.com/cuda-downloads](https://developer.nvidia.com/cuda-downloads).
 
+Software requiremnts:
```

Comment:
If you follow the conda dev environment instructions below, these dependencies are automatically satisfied, as are others (these aren't the only dependencies).  So I'm not sure we want to add this text. @kkraus14 ?

Reply:
But this PR (and the related issue) is not about Conda but building from source with clear documentation.
I can certainly add a phrase about the other dependencies that are getting pulled as submodules.

Reply:
The point is we recommend using conda to satisfy the dependencies when building from source.

Reply:
To be clear, I'm OK with specifying the requirements in this way as long as they appear *after* instructions for using Conda to satisfy requirements.  (And in case it wasn't clear, we recommend using Conda to satisfy requirements even if you are building libcudf from source.)